.github/workflows/release.yaml

```diff
@@ -267,31 +261,66 @@ jobs:
         run: |
           hack/build/ci/install-cgo-dependencies.sh
       - name: Generate release notes
-        id: generate_notes
+        shell: bash
+        env:
+          PRE_RELEASE: ${{ contains(github.ref, '-rc.') }}
         run: |
           hack/build/ci/generate-release-notes.sh
       - name: Generate manifests
         env:
-          IMAGE: ${{ matrix.url }}/${{ secrets[matrix.repository] }}
-          TAG: ${{ needs.prepare.outputs.version }}
-          CHART_VERSION: ${{ needs.prepare.outputs.version }}
-        run: make manifests
-      - name: Rename manifests
+          version: ${{ needs.prepare.outputs.version }}
+          platforms: |
+            kubernetes
+            openshift
+          images: |
+            docker.io/dynatrace/dynatrace-operator
+            registry.connect.redhat.com/dynatrace/dynatrace-operator
+        run: |
+          hack/build/ci/generate-release-manifests.sh "$platforms" "$images" "$version"
+      - name: Rename manifests if pre-release
+        if: ${{ contains(github.ref, '-rc.') }}
         run: |
           mv config/deploy/kubernetes/kubernetes-all.yaml config/deploy/kubernetes/kubernetes.yaml
           mv config/deploy/openshift/openshift-all.yaml config/deploy/openshift/openshift.yaml
-      - name: Generate CRD
-        run: make manifests/crd/release
-      - name: Pre-release
+      - name: Build helm packages
+        uses: ./.github/actions/build-helm
+        with:
+          version_without_prefix: ${{ needs.prepare.outputs.version_without_prefix }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          secring: ${{ secrets.HELM_SECRING }}
+          passphrase: ${{ secrets.HELM_PASSPHRASE }}
+          output-dir: "./helm-pkg"
+      - name: Create pre-release
         uses: softprops/action-gh-release@v1
+        if: ${{ contains(github.ref, '-rc.') }}
```

Comment:
to me it seems splitting this part up into 2 separate would maybe duplicate some logic but would make it easier to read and distinguishable, not really a problem imho but a suggestion from my side. What is you take on that?